File: src/main/java/com/sjs/lootbotga/game/player/Player.java
package com.sjs.lootbotga.game.player;

import com.sjs.lootbotga.game.Table;
import com.sjs.lootbotga.game.cards.Card;

import java.util.List;
import java.util.Map;

public interface Player {
	Integer getId();

	void setId(Integer id);
	
	List<Card> getHand();

	List<Card> getBooty();

	Move planMove(Table table);

	Map<GameState, Move> getStrategy();

	void setStrategy(Map<GameState, Move> strategy);
}

File: src/main/kotlin/com/sjs/lootbotga/game/player/Player.kt
package com.sjs.lootbotga.game.player

import com.sjs.lootbotga.game.Table
import com.sjs.lootbotga.game.cards.Card

interface Player {
    fun getId(): Int?

    fun setId(id: Int?)

    fun getHand(): List<Card?>?

    fun getBooty(): List<Card?>?

    fun planMove(table: Table?): Move?

    fun getStrategy(): Map<GameState?, Move?>?

    fun setStrategy(strategy: Map<GameState?, Move?>?)

    fun setHand(hand: List<Card>)

    fun setBooty(booty: List<Card>)
}